compiler/rustc_resolve/src/effective_visibilities.rs

```diff
@@ -81,7 +81,7 @@ impl<'r, 'a, 'tcx> EffectiveVisibilitiesVisitor<'r, 'a, 'tcx> {
             def_effective_visibilities: Default::default(),
             import_effective_visibilities: Default::default(),
             current_private_vis: Visibility::Restricted(CRATE_DEF_ID),
-            changed: false,
+            changed: true,
```

Comment:
Could you split this into a separate commit (with a test)?
This is a bugfix unrelated to private-in-public lints.

Reply:
Done: https://github.com/rust-lang/rust/pull/112426